webhook/resources/volume/validator.go

```diff
@@ -362,6 +362,10 @@ func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newV
 		return fmt.Errorf("storage class %v of PVC %v does not allow the volume expansion", pvcSCName, pvcName)
 	}
 
+	if (pvcStorageClass.Parameters["fsType"] == "ext4" && newSize >= util.MaxExt4VolumeSize) || (pvcStorageClass.Parameters["fsType"] == "xfs" && newSize >= util.MaxXfsVolumeSize) {
```

Comment:
Should we add a check in Create() function as well?

Reply:
There might be no PV/PVC we can get if we just create a volume from Longhorn UI, so I took the suggestion from Eric https://github.com/longhorn/longhorn/issues/7423#issue-2054290621
```txt
Do not schedule replicas for 16TiB+ volumes to ext4 disks
```

Reply:
I don't think this check is necessary. It is ok to create an ext4 file system that is larger than 16 TiB. We just can't create a file system larger than 16 TiB on top of an ext4 file system (within a 16 TiB file).

Reply:
Removed checking the PVC when expanding volume.

Reply:
What Eric said, it's longhorn FS disk.